cdm/src/main/java/ucar/nc2/iosp/NCheader.java

```diff
@@ -58,41 +74,47 @@
         if(raf.readBytes(magic, 0, MAGIC_NUMBER_LEN) < MAGIC_NUMBER_LEN)
             return 0; // unknown
         // Some version of CDF
-        if(magic[0] == (byte) 'C'
-                && magic[1] == (byte) 'D'
-                && magic[2] == (byte) 'F') {
-            if(magic[3] == 0x01) return NC_FORMAT_CLASSIC;
-            if(magic[3] == 0x02) return NC_FORMAT_64BIT_OFFSET;
-            if(magic[3] == 0x05) return NC_FORMAT_CDF5;
-            return 0; // unknown
+        int hdrlen = 0;
+        byte[] byte4 = Arrays.copyOfRange(magic,0,CDF1HEAD.length);
+        hdrlen = CDF1HEAD.length;
+        format = 0;
+        if(Arrays.equals(CDF1HEAD,byte4))
+            format = NC_FORMAT_CLASSIC;
+        else if(Arrays.equals(CDF2HEAD,byte4))
+            format = NC_FORMAT_64BIT_OFFSET;
+        else if(Arrays.equals(CDF1HEAD,byte4))
+            format = NC_FORMAT_CDF5;
+        if(format != 0) {
+            raf.seek(hdrlen);
+            return format;
         }
         // For HDF5/4, we need to search forward
```

Comment:
According to [the docs](https://support.hdfgroup.org/release4/doc/UsrGuide_html/UG_html-2.htm#906736832_pgfId-92358), the HDF4 magic bytes are always at the beginning of the file. This is in contrast to [HDF5](https://support.hdfgroup.org/HDF5/doc/H5.format.html#Superblock). It looks like we've had code in `H4header` to search forward for the magic since 2007, but it's been unnecessary all that time. So, you could simplify the code here and search for `H4HEAD` in the same way that you search for `CDF1HEAD`, `CDF2HEAD`, and `CDF5HEAD`.